src/Cake.Common/Tools/OctopusDeploy/OctopusDeployRunner.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Linq;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Core.Utilities;
+
+namespace Cake.Common.Tools.OctopusDeploy
+{
+    /// <summary>
+    /// The OctopusDeploy runner
+    /// </summary>
+    public sealed class OctopusDeployRunner : Tool<CreateReleaseSettings>
```

Comment:
Epic contribution! Thank you!

What @RichiCoder1 is saying is correct. A runner should be tightly coupled with specific settings. The thought behind this is to keep the runners specialized, single responsibility and easy to test. I know that there might be different opinions about this (there always is), but I think it's important to be consistent with the other tools like the [NuGet](https://github.com/cake-build/cake/tree/develop/src/Cake.Common/Tools/NuGet) tools.

I think a good idea would be to rename the `OctopusDeployRunner` to something like `OctopusDeployRelaseCreator` or similar that reflect the responsibilities of the tool.

@matdube, @RichiCoder1, @devlead: Any naming suggestions for this?